src/query/sql/src/planner/plans/constant_table_scan.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use common_catalog::table_context::TableContext;
+use common_exception::Result;
+use common_expression::Column;
+use common_expression::DataSchemaRef;
+use itertools::Itertools;
+
+use crate::optimizer::ColumnSet;
+use crate::optimizer::Distribution;
+use crate::optimizer::PhysicalProperty;
+use crate::optimizer::RelExpr;
+use crate::optimizer::RelationalProperty;
+use crate::optimizer::RequiredProperty;
+use crate::optimizer::StatInfo;
+use crate::plans::Operator;
+use crate::plans::RelOp;
+
+// Constant table is a table with constant values.
+#[derive(Clone, Debug)]
+pub struct ConstantTableScan {
+    pub values: Vec<Column>,
+    pub num_rows: usize,
```

Comment:
I'm not sure if the field is necessary, seems we can get it from `values[0].len()`.

Reply:
`num_rows` is needed, as `values` may be empty after `UnusedColumnPruner` optimization.
for example
```sql
mysql> select 1 from (values('a'),('b'),('c'));
+------+
| 1    |
+------+
|    1 |
|    1 |
|    1 |
+------+
```